engine/src/main/java/org/terasology/engine/core/modes/loadProcesses/CreateWorldEntity.java

```diff
@@ -52,64 +64,59 @@ public String getMessage() {
 
     @Override
     public boolean step() {
-        EntityManager entityManager = context.get(EntityManager.class);
-        ChunkProvider chunkProvider = context.get(ChunkProvider.class);
+        this.entityManager = context.get(EntityManager.class);
+        this.worldGenerator = context.get(WorldGenerator.class);
+        this.config = context.get(Config.class);
+        this.chunkProvider = context.get(ChunkProvider.class);
```

Comment:
the plan here is to drive this all through dependency injection. all these loadprocess objects should be transient so they are built used and torn down. not sure when would be a good time to look at gestalt-DI but I think we can preemptively move these to fields and later drop usage of the context when running these steps.

Reply:
we're putting them in fields because it'll use field injection? or constructor injection, which will need to store them in fields somewhere? or does it provide injection on non-constructor methods too?

Reply:
either will work but I would opt for field injection in this case. gestalt-DI will throw an exception if it can't find an object to inject unless you use Optional<>